tstrait/trait_model.py

```diff
@@ -9,139 +9,277 @@ class TraitModel:
 
     :param model_name: Name of the trait model.
     :type model_name: str
-    :param trait_mean: Mean value of the simulated effect sizes.
-    :type trait_mean: float
-    :param trait_var: Variance of the simulated effect sizes.
-    :type trait_var: float
     """
 
-    def __init__(self, model_name, trait_mean, trait_var):
-        self._model_name = model_name
-        self.trait_mean = trait_mean
-        self.trait_var = trait_var
+    def __init__(self, name):
+        self.name = name
 
-    def sim_effect_size(self, num_causal, allele_freq, rng):
+    def _check_parameter(self, num_causal, rng):
+        if not isinstance(num_causal, numbers.Number):
+            raise TypeError("Number of causal sites must be an integer")
+        if int(num_causal) != num_causal or num_causal <= 0:
+            raise ValueError("Number of causal sites must be a positive integer")
+        if not isinstance(rng, np.random.Generator):
+            raise TypeError("rng must be a numpy random generator")
+
+    def frequency_dependence(self, allele_freq, alpha):
```

Comment:
I don't think we're using this function here any more, so it can be removed?

Reply:
I just modified it. Thank you for your suggestion.